quickwit-cli/src/lib.rs

```diff
@@ -654,6 +657,77 @@ impl ThroughputCalculator {
     }
 }
 
+/// Normalizes a URI
+/// A `file://` protocol is assumed if not specified.
+/// File URIs are resolved (normalised) to absolute path.
+/// Handles special path characters like (~, ., ..)
+pub fn normalize_uri(uri: &str) -> anyhow::Result<String> {
+    let parts: Vec<&str> = uri.split("://").collect();
+    let (protocol, mut path) = match parts.len() {
+        1 => (FILE_PROTOCOL, String::from(parts[0])),
+        _ => (
+            parts[0],
+            parts
+                .into_iter()
+                .skip(1)
+                .map(String::from)
+                .collect::<Vec<_>>()
+                .join(""),
+        ),
+    };
+
+    if protocol == FILE_PROTOCOL {
+        if path.starts_with('~') {
+            let home_dir_path = home::home_dir()
+                .context("Could nto get home directory.")?
+                .to_string_lossy()
+                .to_string();
+
+            path.replace_range(0..1, &home_dir_path);
+        }
+
+        if !path.starts_with('/') {
+            path.insert(0, '/');
```

Comment:
for file://my_path I agree resolving from root might be a good idea. But otherwise resolving from CWD might be a better spec.

Reply:
I have switched to `CWD`  given we have a CLI tool and that's what I would expect as an end user. Let's see what @guilload thinks about it.